packages/stargate/src/queries/bank.ts

```diff
@@ -0,0 +1,57 @@
+import { Bech32, toAscii } from "@cosmjs/encoding";
+
+import { cosmos } from "../generated/codecimpl";
+import { QueryClient } from "../queryclient";
+
+/**
+ * Use this to convert a protobuf.js class to the interface (e.g. Coin to ICoin)
+ * in a ways that makes Jasmine's toEqual happy.
+ */
+function toObject<I extends object>(thing: I): Omit<I, never> {
+  return { ...thing };
+}
+
+export interface BankExtension {
+  readonly bank: {
+    readonly balance: (address: string, denom: string) => Promise<cosmos.ICoin | null>;
+    readonly unverified: {
+      readonly allBalances: (address: string) => Promise<cosmos.ICoin[]>;
+    };
+  };
+}
+
+export function setupBankExtension(base: QueryClient): BankExtension {
+  // Use this service to get easy typed access to query methods
+  // This cannot be used to for proof verification
+  const queryService = cosmos.bank.Query.create((method: any, requestData, callback) => {
```

Comment:
The autogenerated class `cosmos.bank.Query` contains all the query methods including types, request encoding and response decoding.